sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreResultDiagnostics.java

```diff
@@ -151,6 +148,28 @@ public String getStorePhysicalAddressEscapedPath() {
             : null;
     }
 
+    public String[] getPartitionAndReplicaId() {
+        return getPartitionAndReplicaId(this.getStorePhysicalAddressEscapedPath());
+    }
+
+    public static String[] getPartitionAndReplicaId(String serviceAddress) {
+        if (serviceAddress == null) {
+            return ArrayUtils.EMPTY_STRING_ARRAY;
+        }
+
+        String[] serviceAddressParts = serviceAddress.split("/");
+        // Sample value for serviceAddress
+        // /apps/f88bfdf4-2954-4324-aad3-f1686668076d/services/3359112a-719d-474e-aa51-e89a142ae1b3/partitions/512fe816-24fa-4fbb-bbb1-587d2ce19851/replicas/133038444008943156p/
+        if (serviceAddressParts.length != 9) {
```

Comment:
opt: may be ok if in future there are 10 or 11 parts? I.e. check if there are min 9 parts at least but more can be ok.
Your call on this.